addons/pos_self_order/controllers/ordering.py

```diff
@@ -0,0 +1,455 @@
+                #  FIXME: find out what pricelist_id to use
+                #  'pricelist_id': 1,
+                "partner_id": False,
+                "user_id": request.session.uid,
+                # we only need the digits of the order_id, not the whole id, which starts with word "Order"
+                "uid": order.get("id")[6:],
+                "sequence_number": order.get("sequence_number"),
+                "creation_date": str(fields.Datetime.now()),
+                "fiscal_position_id": False,
+                "to_invoice": False,
+                "to_ship": False,
+                "is_tipped": False,
+                "tip_amount": 0,
+                "access_token": order.get("access_token"),
+                "server_id": False,
+                "table_id": order.get("table_id"),
+            },
+            "to_invoice": False,
+            "session_id": order.get("session_id"),
+        }
+
+    def _compute_amount_total(self, lines: List[Dict]) -> float:
+        return sum(orderline[2].get("price_subtotal_incl") for orderline in lines)
+
+    def _compute_amount_tax(self, lines: List[Dict]) -> float:
+        return sum(
+            orderline[2].get("price_subtotal_incl") - orderline[2].get("price_subtotal")
+            for orderline in lines
+        )
+
+    # the 2nd argument of order_id is the login number;
+    # the regular pos receives the login number from the backend on the first load
+    # the problem is that if the pos has login number 1, for example
+    # and we create a new order in the database, with sequence number 1, for example,
+    # when the regular pos will try to create a new order, it will also use sequence number 1
+    # and login number 1 and the new order will have the same id as the order created by the self order app
+    # so it will ultimately not be recorded in the database.
+    # ( the regular pos does not check what was the last order id from the database )
+    # it keeps track itself of the last order id -- this is why this problem occurs
+    # I thought that it is best to make the pos_self_order app work around
+    # the problem, instead of modifying the regular pos.
+    # so I decided to use the numbers from 900 to 999 for the login number
+    # this way, the regular pos will  have a login number of 1, 2, 3, etc
+    # and the pos_self_order app will have a login number of 901, 902, 903, etc,
+    # where 901 is the login number of the table with id = 1,
+    # 902 is the login number of the table with id = 2, etc
+    # TODO: make it so the regular pos can't have a login number of 900 or more
+    def _generate_unique_id(self, pos_session_id: int, table_id: int, sequence_number: int) -> str:
+        """
+        Generates a public identification number for the order.
+        :return: The order name. Example: Order 00001-001-0001
+        """
+        return f"Order {pos_session_id:0>5}-{900 + table_id :0>3}-{sequence_number:0>4}"
```

Comment:
the problem that i see here is that the pos does not check with the backend when it takes a new login number. 
So if we look in the backend and see that the last login number is 5, for ex, and we start using `6`, if the pos reloads it will start using the number `6` too.

Reply:
There could be ways to synchronize the pos_reference. Perhaps we open a session that's only exclusive for self-ordering and there, we can query the server for the pos_reference.